inst/include/dust/xoshiro.hpp

```diff
@@ -42,6 +42,8 @@ class Xoshiro {
   void jump();
   void long_jump();
 
+  void get_state(std::vector<uint64_t>& state);
```

Comment:
Question rather than review comment: why do all the Dust functions return void, and we provide a vector of the correct size to insert results into? That feels quite C rather than C++ to me. Does it make it easier with the R interface/pointers? (perhaps I am just forgetting something we discussed when we first wrote it...)

Reply:
if we were inserting into it we'd have to pass in the index as well - here're we're just appending. If I was doing it in C I'd probably do as you describe though and make this

```c
size_t get_state(uint64_t *state, size_t index) {
  for (size_t i = 0; i < XOSHIRO_WIDTH; ++i, ++index) {
    state[index] = _state[i];
  }
  return index;
}
```